src/results.cpp

```diff
@@ -134,6 +155,10 @@ size_t Results::size()
         case Mode::LinkView: return m_link_view->size();
         case Mode::Query:
             m_query.sync_view_if_needed();
+            if (m_distinct) {
```

Comment:
`if (!m_distinct) return m_query.count(); REALM_FALLTHROUGH;` avoids duplicating the (admittedly trivial) logic for the tableview case.